timber-lint/src/main/java/timber/lint/WrongTimberUsageDetector.java

```diff
@@ -49,156 +45,103 @@
-        VariableReference ref = (VariableReference) expression;
-        if (!"Timber".equals(ref.astIdentifier().astValue())) {
-          return;
-        }
-      }
-
-      if (node.astArguments().isEmpty()) {
-        return;
-      }
-      Node argument = node.astArguments().iterator().next();
-      String tag = findLiteralValue(context, argument);
-      if (tag != null && tag.length() > 23) {
-        String message = String.format(
-            "The logging tag can be at most 23 characters, was %1$d (%2$s)",
-            tag.length(), tag);
-        context.report(ISSUE_TAG_LENGTH, node, context.getLocation(argument), message);
-      }
-    } else if (node.astOperand() instanceof VariableReference) {
-      VariableReference ref = (VariableReference) node.astOperand();
-      if ("Log".equals(ref.astIdentifier().astValue())) {
-        context.report(ISSUE_LOG, node, context.getRangeLocation(node, 0, node.astName(), 0),
-            "Using 'Log' instead of 'Timber'");
-        return;
-      }
-      if (!"Timber".equals(ref.astIdentifier().astValue())) {
-        return;
-      }
-      checkThrowablePosition(context, node);
-      checkArguments(context, node);
-    } else if (isAstOperandTimberTagLogPattern(node)) {
-      List<Node> siblings = node.astOperand().getParent().getChildren();
-
-      List<Node> logNodes = siblings.subList(1, siblings.size());
-      List<Expression> expressionNodes = new ArrayList<>(); // casted version of logNodes
-
-      List<Node> logArgs = logNodes.subList(1, logNodes.size());
-      for (Node n : logArgs) {
-        if (!(n instanceof lombok.ast.Expression)) {
-          continue; // this is a failure; see `if` guard just outside of this loop
-        }
-        expressionNodes.add((Expression) n);
-      }
-
-      // If we're short an Expression, then our guess about this node and its siblings may be wrong,
-      if (expressionNodes.size() == logArgs.size()) {
-        checkStringFormatArguments(
-            context, node, expressionNodes.iterator(), expressionNodes.size());
-      }
+  @Override public void visitMethod(JavaContext context, JavaElementVisitor visitor,
+      PsiMethodCallExpression call, PsiMethod method) {
+    PsiReferenceExpression methodExpression = call.getMethodExpression();
+    String fullyQualifiedMethodName = methodExpression.getQualifiedName();
+    if ("java.lang.String.format".equals(fullyQualifiedMethodName)) {
+      checkNestedStringFormat(context, call);
+      return;
+    }
+    if (fullyQualifiedMethodName.startsWith("timber.log.Timber.tag")) {
+      checkTagLength(context, call);
+      return;
+    }
+    if (fullyQualifiedMethodName.startsWith("android.util.Log.")) {
+      context.report(ISSUE_LOG, methodExpression, context.getLocation(methodExpression),
+          "Using 'Log' instead of 'Timber'");
+      return;
+    }
+    // handles Timber.X(..) and Timber.tag(..).X(..) where X in (v|d|i|w|e|wtf)
```

Comment:
nit: Capitalize "handles" and add a period. [Comments should be sentences][1]

 [1]: http://nedbatchelder.com/blog/201401/comments_should_be_sentences.html